assets/js/modules/analytics-4/datastore/report.js

```diff
@@ -154,6 +158,104 @@ const baseSelectors = {
 
 		return reports[ stringifyObject( options ) ];
 	},
+
+	/**
+	 * Determines whether the Analytics 4 module is still gathering data.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {boolean|undefined} Returns `true` if gathering data, otherwise `false`. Returns `undefined` while resolving.
+	 */
+	isGatheringData: createRegistrySelector( ( select ) => () => {
```

Comment:
A note to the code reviewer: Please remember, a lot of the code here is copied and modified to work with GA4. Therefore it's probably best not to treat this as a complete "root and branch" review, in that we want to remain consistent with the existing code (unless there's a compelling reason to change something). It might be beneficial to look at the commit history, I've tried to keep a clean separation between the copying and modifying stages.